### ISSUE TITLE
Saving pre_params in train.lua

### ISSUE THREAD

**Issue by harkmug** [johndun/mnist-torch7#1]
In line 27 of train.lua, should `torch.save('models/preprocessing_params.t7', params)` have `pre_params` instead of `params`? I was trying to run predict on the trained model for a new dataset (the Kaggle MNIST Digit Recognition challenge). Are you thinking of importing the `predict.lua`  code from [here](https://github.com/nagadomi/kaggle-cifar10-torch7/blob/cuda-convnet2/predict.lua). Thanks for the code!

**Comment by johndun** [comment]
Definitely should be pre_params. I had spent most of my time working with
the code in validate.lua.

I wasn't aware of a Kaggle MNIST dataset. I'd imagine you'd want to do a
bunch of jittering to place on the leaderboard.

I'll have to check it out. Thanks.

**Comment by harkmug** [comment]
Thanks, John. I got a predict function working and ran the Kaggle MNIST test set through the "setup_mlp" model (changing the input image size to 28 x 28 before training). It made it into the top 10 with score of 0.99743! Next up, I'll try the convnet1.

(This is raghu, btw)

**Comment by johndun** [comment]
Did you train using the data from torch7 demo? 0.99743 is too high for a simple MLP model. I have to do some crazy stuff to get to 0.9965 on the standard train/test split. See [here](http://rodrigob.github.io/are_we_there_yet/build/classification_datasets_results.html). I think Kaggle must have shuffled the test set around.

**Comment by harkmug** [comment]
Yes, it is cheating a bit. Kaggle split the MNIST data set into 42,000 training and 28,000 test. But I did not use their training data as you had the full mnist data already loaded as part of this demo. I just used the trained model to predict on the 28K test set. Presumably, many of the same images were already in the demo training data used by the model). Hence the high scores for all participants (Most of the 99.5+ folks must be using the full mnist dataset not just the 42K provided). I was just trying this out as a way to get familiar with the demo code.

**Comment by johndun** [comment]
I created a [new repository](https://github.com/johndun/kaggle-mnist) with some better code for fitting and testing against the Kaggle version. The same 1 hidden layer model gets to 0.97957 on the public leaderboard test set (127th position when I submitted) when only trained on the 42k samples in the Kaggle training set.
